arch/posix/CMakeLists.txt

```
@@ -133,7 +133,13 @@ if (CONFIG_GPROF)
   target_link_options(native_simulator INTERFACE "-pg")
 endif()
 
-zephyr_compile_definitions(_POSIX_C_SOURCE=200809 _XOPEN_SOURCE=600 _XOPEN_SOURCE_EXTENDED)
+zephyr_compile_options(
+  -U_POSIX_C_SOURCE
+  -D_POSIX_C_SOURCE=200809L
+  -U_XOPEN_SOURCE
+  -D_XOPEN_SOURCE=600
+  _XOPEN_SOURCE_EXTENDED
+)
```

`_XOPEN_SOURCE_EXTENDED` is being treated as a linker file path in `zephyr_compile_options`:

```suggestion
)
zephyr_compile_definitions(_XOPEN_SOURCE_EXTENDED)
```